### PR TITLE
Add `ManyToManyField.m2m_field_name` and `Model._get_pk_val` methods

### PR DESCRIPTION
**Pull request by YPCrumble**
Thanks for maintaining this library! This PR adds a couple methods I use in my codebase that are part of models.Model and ManyToManyField. Let me know if any questions or concerns?

# I have made things!

## Related issues

Refs #204

### CODE REVIEW

File: django-stubs/db/models/base.pyi [hunk 1]
```diff
@@ -99,5 +99,6 @@ class Model(metaclass=ModelBase):
     @classmethod
     def check(cls, **kwargs: Any) -> list[CheckMessage]: ...
     def __getstate__(self) -> dict: ...
+    def _get_pk_val(self, meta: Options[Any] = ...) -> str: ...
```
Comment (Viicos) [inline django-stubs/db/models/base.pyi]:
```suggestion
    def _get_pk_val(self, meta: Options[Self] = ...) -> str: ...
```

Probably?

**Comment by YPCrumble** [comment]
Thank you @Viicos  - updated!

File: django-stubs/db/models/base.pyi [hunk 2]
```diff
@@ -99,5 +99,6 @@ class Model(metaclass=ModelBase):
     @classmethod
     def check(cls, **kwargs: Any) -> list[CheckMessage]: ...
     def __getstate__(self) -> dict: ...
+    def _get_pk_val(self, meta: Options[Self] = ...) -> str: ...
```
Comment (intgr) [inline django-stubs/db/models/base.pyi]:
```suggestion
    def _get_pk_val(self, meta: Options[Self] | None = None) -> str: ...
```

**Review by intgr — COMMENTED** [review]
Thanks! Have a look at stubtest output too.

You can run it in a virtualenv locally with `bash ./_local/stubtest.sh`

**Review by intgr — APPROVED** [review]
Thanks! I pushed fixes the allowlist as well to make CI pass (and move around related allowlist entries).

**Pull request merged**